src/controllers/cats.ts

```diff
@@ -0,0 +1,57 @@
+import { Request, Response, NextFunction } from 'express';
+import validator from '../utils/validator';
+import * as cats from '../models/cats';
+import { type User } from '@prisma/client';
+
+export function createCat(
+	user: User,
+	request: Request,
+	response: Response,
+	next: NextFunction
+) {
+	const schema = {
+		name: 'string,optional',
+		description: 'string,optional',
+		picture_url: 'string,optional',
+	};
+	const payload = request.body;
+	const result = validator(payload, schema);
+
+	if (!result.success) return next({ status: 400, message: result.errors });
+
+	cats.createCat(user.username, result.body).then((cat) => {
```

Comment:
Should this be user's id (generated by us) not username (generated by them) to guard against corrupting inputs?

Reply:
This was pre-middleware that I forgot to change fully. Will work on it

Reply:
Although in theory there should never be a case where a username should be able to corrupt this - If that was true, we have a bigger issue on our hands. Still gonna change over to a new method tho